crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -51,10 +69,10 @@ def sync
   #
   # Provides the restful api call for
   # Activate a Repository   /utils/repositories/activate   POST  Creates Repository DataBagItem
-  # required parameters: platform, repo
+  # required parameters: platform, arch, repo
   def activate
-    return render_not_found if params[:platform].nil? || params[:repo].nil?
-    ret, _message = ProvisionerService.new(logger).enable_repository(params[:platform], params[:repo])
+    return render_not_found if params[:platform].nil? || params[:arch].nil? || params[:repo].nil?
+    ret, _message = ProvisionerService.new(logger).enable_repository(params[:platform], params[:arch], params[:repo])
```

Comment:
Metrics/LineLength: Line is too long. [117/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)